src/Core/AbstractTest.php

```diff
@@ -182,6 +182,30 @@ public function setPending($state)
     }
 
     /**
+     * Set the focused status of the test and its children according to the
+     * supplied focus pattern and/or skip pattern
+     *
+     * @param string|null $focusPattern
+     * @param string|null $skipPattern
+     */
+    public function applyFocusPatterns($focusPattern, $skipPattern = null)
+    {
+        $title = $this->getTitle();
+
+        if ($skipPattern !== null && preg_match($skipPattern, $title)) {
```

Comment:
Should we validate patterns somehow? If a non delimited regex is given, PHP will throw warnings:

```
$ php -a
php > preg_match('lol', 'rofl');
PHP Warning:  preg_match(): Delimiter must not be alphanumeric or backslash in php shell code on line 1

Warning: preg_match(): Delimiter must not be alphanumeric or backslash in php shell code on line 1
```